schema-registry/serdes/src/main/java/com/hortonworks/registries/schemaregistry/serdes/avro/AvroSerDesHandler.java

```diff
@@ -25,10 +25,13 @@
  */
 public interface AvroSerDesHandler {
 
-    void handlePayloadSerialization(OutputStream outputStream, Object input);
+    void handlePayloadSerialization(OutputStream outputStream,
+                                    Object input,
+                                    DatumWriterProvider datumWriterProvider);
 
     Object handlePayloadDeserialization(InputStream payloadInputStream,
                                         Schema writerSchema,
                                         Schema readerSchema,
-                                        boolean useSpecificAvroReader);
+                                        DatumReaderProvider datumReaderProvider);
```

Comment:
Same as earlier about the backward incompatible change. You can add a new method here to take custom `DatumReaderProvider `.

Reply:
Sure will do.